### ISSUE TITLE
Building examples in documentation fails with recent Jupyter notebook versions.

### ISSUE THREAD

**Issue by jgosmann** [nengo/nengo#1308]
This is the exception I get ([note that I had to remove the conversion to the warning to get the actual backtrace](https://github.com/nengo/nengo/blob/master/nengo/utils/docutils.py#L105) which should probably fixed to):
```
# Sphinx version: 1.5.5
# Python version: 3.5.3 (CPython)
# Docutils version: 0.13.1 release
# Jinja2 version: 2.9.6
# Last messages:
#   Running Sphinx v1.5.5
#   loading pickled environment...
#   not yet created
#   loading intersphinx inventory from http://docs.scipy.org/doc/numpy/objects.inv...
#   intersphinx inventory has moved: http://docs.scipy.org/doc/numpy/objects.inv -> https://docs.scipy.org/doc/numpy/objects.inv
#   building [mo]: all of 0 po files
#   building [html]: all source files
#   updating environment:
#   10 added, 0 changed, 0 removed
#   reading sources... [ 10%] examples/associative_memory
# Loaded extensions:
#   sphinx.ext.todo (1.5.5) from /Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/sphinx/ext/todo.py
#   sphinx.ext.viewcode (1.5.5) from /Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/sphinx/ext/viewcode.py
#   nengo.utils.docutils (unknown version) from /Volumes/Home/blubb/Documents/projects/uni/nengo/nengo/utils/docutils.py
#   alabaster (0.7.10) from /Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/alabaster/__init__.py
#   sphinx.ext.intersphinx (1.5.5) from /Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/sphinx/ext/intersphinx.py
#   numpydoc (unknown version) from /Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/numpydoc/__init__.py
#   sphinx.ext.mathjax (1.5.5) from /Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/sphinx/ext/mathjax.py
#   sphinx.ext.autodoc (1.5.5) from /Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/sphinx/ext/autodoc.py
#   sphinx.ext.autosummary (1.5.5) from /Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/sphinx/ext/autosummary/__init__.py
Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/ipython_genutils/ipstruct.py", line 132, in __getattr__
    result = self[key]
KeyError: 'worksheets'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/sphinx/setup_command.py", line 176, in run
    app.build(force_all=self.all_files)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/sphinx/application.py", line 327, in build
    self.builder.build_all()
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/sphinx/builders/__init__.py", line 211, in build_all
    self.build(None, summary='all source files', method='all')
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/sphinx/builders/__init__.py", line 265, in build
    self.doctreedir, self.app))
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/sphinx/environment/__init__.py", line 556, in update
    self._read_serial(docnames, app)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/sphinx/environment/__init__.py", line 576, in _read_serial
    self.read_doc(docname, app)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/sphinx/environment/__init__.py", line 684, in read_doc
    pub.publish()
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/core.py", line 217, in publish
    self.settings)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/sphinx/io.py", line 55, in read
    self.parse()
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/readers/__init__.py", line 78, in parse
    self.parser.parse(self.input, document)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/parsers/rst/__init__.py", line 185, in parse
    self.statemachine.run(inputlines, document, inliner=self.inliner)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/parsers/rst/states.py", line 170, in run
    input_source=document['source'])
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/statemachine.py", line 239, in run
    context, state, transitions)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/statemachine.py", line 460, in check_line
    return method(match, context, next_state)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/parsers/rst/states.py", line 2980, in text
    self.section(title.lstrip(), source, style, lineno + 1, messages)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/parsers/rst/states.py", line 326, in section
    self.new_subsection(title, lineno, messages)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/parsers/rst/states.py", line 394, in new_subsection
    node=section_node, match_titles=True)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/parsers/rst/states.py", line 281, in nested_parse
    node=node, match_titles=match_titles)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/parsers/rst/states.py", line 195, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/statemachine.py", line 239, in run
    context, state, transitions)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/statemachine.py", line 460, in check_line
    return method(match, context, next_state)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/parsers/rst/states.py", line 2318, in explicit_markup
    nodelist, blank_finish = self.explicit_construct(match)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/parsers/rst/states.py", line 2330, in explicit_construct
    return method(self, expmatch)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/parsers/rst/states.py", line 2073, in directive
    directive_class, match, type_name, option_presets)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/parsers/rst/states.py", line 2122, in run_directive
    result = directive_instance.run()
  File "/Volumes/Home/blubb/Documents/projects/uni/nengo/nengo/utils/docutils.py", line 107, in run
    nb, dest_path, skip_exceptions=skip_exceptions)
  File "/Volumes/Home/blubb/Documents/projects/uni/nengo/nengo/utils/ipython.py", line 262, in export_evaluated
    nb_runner.run_notebook(skip_exceptions=skip_exceptions)
  File "/Volumes/Home/blubb/Documents/projects/uni/nengo/nengo/utils/ipython.py", line 395, in run_notebook
    for i, cell in enumerate(self.iter_code_cells()):
  File "/Volumes/Home/blubb/Documents/projects/uni/nengo/nengo/utils/ipython.py", line 384, in iter_code_cells
    for ws in self.nb.worksheets:
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/ipython_genutils/ipstruct.py", line 134, in __getattr__
    raise AttributeError(key)
AttributeError: worksheets
```

This particular error is fixable with [this code](https://github.com/ctn-waterloo/devscripts/blob/master/clearoutputs.py#L16) from the devscripts. However, than I still get the following exception:
```
# Sphinx version: 1.5.5
# Python version: 3.5.3 (CPython)
# Docutils version: 0.13.1 release
# Jinja2 version: 2.9.6
# Last messages:
#   Running Sphinx v1.5.5
#   loading pickled environment...
#   not yet created
#   loading intersphinx inventory from http://docs.scipy.org/doc/numpy/objects.inv...
#   intersphinx inventory has moved: http://docs.scipy.org/doc/numpy/objects.inv -> https://docs.scipy.org/doc/numpy/objects.inv
#   building [mo]: all of 0 po files
#   building [html]: all source files
#   updating environment:
#   10 added, 0 changed, 0 removed
#   reading sources... [ 10%] examples/associative_memory
# Loaded extensions:
#   alabaster (0.7.10) from /Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/alabaster/__init__.py
#   sphinx.ext.todo (1.5.5) from /Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/sphinx/ext/todo.py
#   sphinx.ext.autosummary (1.5.5) from /Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/sphinx/ext/autosummary/__init__.py
#   nengo.utils.docutils (unknown version) from /Volumes/Home/blubb/Documents/projects/uni/nengo/nengo/utils/docutils.py
#   sphinx.ext.viewcode (1.5.5) from /Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/sphinx/ext/viewcode.py
#   sphinx.ext.mathjax (1.5.5) from /Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/sphinx/ext/mathjax.py
#   numpydoc (unknown version) from /Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/numpydoc/__init__.py
#   sphinx.ext.intersphinx (1.5.5) from /Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/sphinx/ext/intersphinx.py
#   sphinx.ext.autodoc (1.5.5) from /Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/sphinx/ext/autodoc.py
Traceback (most recent call last):
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/sphinx/setup_command.py", line 176, in run
    app.build(force_all=self.all_files)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/sphinx/application.py", line 327, in build
    self.builder.build_all()
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/sphinx/builders/__init__.py", line 211, in build_all
    self.build(None, summary='all source files', method='all')
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/sphinx/builders/__init__.py", line 265, in build
    self.doctreedir, self.app))
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/sphinx/environment/__init__.py", line 556, in update
    self._read_serial(docnames, app)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/sphinx/environment/__init__.py", line 576, in _read_serial
    self.read_doc(docname, app)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/sphinx/environment/__init__.py", line 684, in read_doc
    pub.publish()
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/core.py", line 217, in publish
    self.settings)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/sphinx/io.py", line 55, in read
    self.parse()
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/readers/__init__.py", line 78, in parse
    self.parser.parse(self.input, document)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/parsers/rst/__init__.py", line 185, in parse
    self.statemachine.run(inputlines, document, inliner=self.inliner)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/parsers/rst/states.py", line 170, in run
    input_source=document['source'])
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/statemachine.py", line 239, in run
    context, state, transitions)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/statemachine.py", line 460, in check_line
    return method(match, context, next_state)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/parsers/rst/states.py", line 2980, in text
    self.section(title.lstrip(), source, style, lineno + 1, messages)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/parsers/rst/states.py", line 326, in section
    self.new_subsection(title, lineno, messages)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/parsers/rst/states.py", line 394, in new_subsection
    node=section_node, match_titles=True)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/parsers/rst/states.py", line 281, in nested_parse
    node=node, match_titles=match_titles)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/parsers/rst/states.py", line 195, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/statemachine.py", line 239, in run
    context, state, transitions)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/statemachine.py", line 460, in check_line
    return method(match, context, next_state)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/parsers/rst/states.py", line 2318, in explicit_markup
    nodelist, blank_finish = self.explicit_construct(match)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/parsers/rst/states.py", line 2330, in explicit_construct
    return method(self, expmatch)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/parsers/rst/states.py", line 2073, in directive
    directive_class, match, type_name, option_presets)
  File "/Volumes/Home/blubb/.virtualenvs/nengodev/lib/python3.5/site-packages/docutils/parsers/rst/states.py", line 2122, in run_directive
    result = directive_instance.run()
  File "/Volumes/Home/blubb/Documents/projects/uni/nengo/nengo/utils/docutils.py", line 107, in run
    nb, dest_path, skip_exceptions=skip_exceptions)
  File "/Volumes/Home/blubb/Documents/projects/uni/nengo/nengo/utils/ipython.py", line 262, in export_evaluated
    nb_runner.run_notebook(skip_exceptions=skip_exceptions)
  File "/Volumes/Home/blubb/Documents/projects/uni/nengo/nengo/utils/ipython.py", line 401, in run_notebook
    self.run_cell(cell)
  File "/Volumes/Home/blubb/Documents/projects/uni/nengo/nengo/utils/ipython.py", line 319, in run_cell
    self.shell.execute(cell.input)
AttributeError: 'ZMQSocketChannel' object has no attribute 'execute'
```

**Comment by tbekolay** [comment]
Yes, I sunk a few weeks into this a few months ago in [this branch](https://github.com/nengo/nengo/tree/nbsphinx). Since it was holding up a release, I opted instead to install old versions of things to build the docs and have been doing it every since. That branch relies on an external package, [nbsphinx](https://github.com/spatialaudio/nbsphinx), which I made some [PRs for](https://github.com/spatialaudio/nbsphinx/pull/33), which also contributed to opting for using old packages for the time being.

**Comment by jgosmann** [comment]
I believe this has been fixed with #1349.
